data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataimport/controlfile/ControlFileValidator.java

```diff
@@ -0,0 +1,225 @@
+      if (!columnNames.contains(mapping.getTargetColumn())) {
+        throw new ControlFileValidationException(
+            CoreError.DATA_LOADER_TARGET_COLUMN_NOT_FOUND.buildMessage(
+                mapping.getTargetColumn(), mapping.getSourceField(), lookupKey));
+      }
+    }
+  }
+
+  /**
+   * Check that the required partition keys are mapped in the control file. Ran only for control
+   * file validation mode KEYS and FULL.
+   *
+   * @param tableMetadata Metadata for one ScalarDB table
+   * @param mappedTargetColumns Set of target columns that are mapped in the control file
+   * @param controlFileTable Control file entry for one ScalarDB table
+   * @throws ControlFileValidationException when a partition key is not mapped
+   */
+  private static void checkPartitionKeys(
+      TableMetadata tableMetadata,
+      Set<String> mappedTargetColumns,
+      ControlFileTable controlFileTable)
+      throws ControlFileValidationException {
+    LinkedHashSet<String> partitionKeyNames = tableMetadata.getPartitionKeyNames();
+    for (String partitionKeyName : partitionKeyNames) {
+      if (!mappedTargetColumns.contains(partitionKeyName)) {
+        throw new ControlFileValidationException(
+            CoreError.DATA_LOADER_MISSING_PARTITION_KEY.buildMessage(
+                partitionKeyName, TableMetadataUtil.getTableLookupKey(controlFileTable)));
+      }
+    }
+  }
+
+  /**
+   * Check that the required clustering keys are mapped in the control file. Ran only for control
+   * file validation mode KEYS and FULL.
+   *
+   * @param tableMetadata Metadata for one ScalarDB table
+   * @param mappedTargetColumns Set of target columns that are mapped in the control file
+   * @param controlFileTable Control file entry for one ScalarDB table
+   * @throws ControlFileValidationException when a clustering key is not mapped
+   */
+  private static void checkClusteringKeys(
+      TableMetadata tableMetadata,
+      Set<String> mappedTargetColumns,
+      ControlFileTable controlFileTable)
+      throws ControlFileValidationException {
+    LinkedHashSet<String> clusteringKeyNames = tableMetadata.getClusteringKeyNames();
+    for (String clusteringKeyName : clusteringKeyNames) {
+      if (!mappedTargetColumns.contains(clusteringKeyName)) {
+        throw new ControlFileValidationException(
+            CoreError.DATA_LOADER_MISSING_CLUSTERING_KEY.buildMessage(
+                clusteringKeyName, TableMetadataUtil.getTableLookupKey(controlFileTable)));
+      }
+    }
+  }
+
+  /**
+   * Check that a control file table mapping does not contain duplicate mappings for the same target
+   * column
+   *
+   * @param controlFileTable Control file entry for one ScalarDB table
+   * @return Set of uniquely mapped target columns
+   * @throws ControlFileValidationException when a duplicate mapping is found
+   */
+  private static Set<String> checkDuplicateColumnMappings(ControlFileTable controlFileTable)
```

Comment:
I think this function name should be `getTargetColumnSet` or something since it converts the data structure not only checking. What do you think?